### PR TITLE
update dependencies & use ci for unit testing

### PR DESCRIPTION
**Pull request by commonism**
the ci changes may require … the usual … iterations

### CODE REVIEW

**Review by noahkw — APPROVED** [review]
I can't currently run most of the tests on my Windows machine but the code changes lgtm.

File: .github/workflows/codecov.yml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    - name: create .env file
+      shell: bash
+      run: |
+        cat <<EOF>.env
+        ACME_TESTS_PG_PW=acme-broker-debug-pw
+        ACME_SUPERUSER_PW=YOUR_SUPERUSER_PW
+        ACME_ADMIN_PW=YOUR_ADMIN_PW
+        ACME_RW_PW=YOUR_READ_WRITE_PW
+        ACME_RO_PW=YOUR_READ_ONLY_PW
+        ACME_PREFIX=/FIXME
```
Comment (noahkw) [inline .github/workflows/codecov.yml:29]: /home/acme ?

**Review by noahkw — COMMENTED** [review]
We may need a small section in the dev docs on how to install and where to put the third party ACME clients to test against. (since they've been removed from the dependencies)

File: .github/workflows/codecov.yml (hunk 1 above)
Comment (commonism) [reply .github/workflows/codecov.yml:29]: currently ACME_PREFIX is not used in the CI - it relates to the docker-compose.yml, will remove.

**Pull request merged**